flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/ZooKeeperCompletedCheckpointStoreTest.java

```diff
@@ -180,17 +89,19 @@ private void testDownloadInternal(
                     }
                 };
 
-        CompletedCheckpointStore store =
-                new DefaultCompletedCheckpointStore<>(
-                        10,
-                        checkpointsInZooKeeper,
-                        zooKeeperCheckpointStoreUtil,
-                        Executors.directExecutor());
+        checkpointsInZk.add(
+                createHandle(
+                        1,
+                        id -> {
+                            throw new ExpectedTestException();
+                        }));
         try {
-            test.accept(store, checkpointsInZk, sharedStateRegistry);
-        } finally {
-            store.shutdown(JobStatus.FINISHED, new CheckpointsCleaner());
-            sharedStateRegistry.close();
+            DefaultCompletedCheckpointStoreUtils.retrieveCompletedCheckpoints(
+                    checkpointsInZooKeeper, zooKeeperCheckpointStoreUtil);
+        } catch (Exception exception) {
+            findThrowable(exception, ExpectedTestException.class)
+                    .ifPresent(ExceptionUtils::rethrow);
+            rethrow(exception);
```

Comment:
You remove the `expected = ` and use `FlinkMatchers.containsCause` instead. This might be a bit simpler.

Reply:
fixed